kornia/feature/siftdesc.py

```diff
@@ -18,154 +20,139 @@ def get_sift_pooling_kernel(ksize:int = 25)-> torch.Tensor:
-             '(' + 'num_ang_bins=' + str(self.num_ang_bins) +\
-             ', ' + 'num_spatial_bins=' + str(self.num_spatial_bins) +\
-             ', ' + 'patch_size=' + str(self.patch_size) +\
-             ', ' + 'rootsift=' + str(self.rootsift) +\
-             ', ' + 'sigma_type=' + str(self.sigma_type) +\
-             ', ' + 'mask_type=' + str(self.mask_type) +\
-             ', ' + 'clipval=' + str(self.clipval) + ')'
+    def __repr__(self) -> str:
+        return self.__class__.__name__ +\
+            '(' + 'num_ang_bins=' + str(self.num_ang_bins) +\
+            ', ' + 'num_spatial_bins=' + str(self.num_spatial_bins) +\
+            ', ' + 'patch_size=' + str(self.patch_size) +\
+            ', ' + 'rootsift=' + str(self.rootsift) +\
+            ', ' + 'clipval=' + str(self.clipval) + ')'
+
     def __init__(self,
-                 patch_size = 41, 
-                 num_ang_bins = 8,
-                 num_spatial_bins = 4,
-                 clipval = 0.2,
-                 rootsift = True,
-                 mask_type = 'Gauss',
-                 sigma_type = 'vlfeat'):
+                 patch_size: int = 41,
+                 num_ang_bins: int = 8,
+                 num_spatial_bins: int = 4,
+                 clipval: float = 0.2,
+                 rootsift: bool = True) -> None:
         super(SIFTDescriptor, self).__init__()
         self.eps = 1e-10
         self.num_ang_bins = num_ang_bins
         self.num_spatial_bins = num_spatial_bins
         self.clipval = clipval
         self.rootsift = rootsift
-        self.mask_type = mask_type
         self.patch_size = patch_size
-        self.sigma_type = sigma_type
-        if self.mask_type == 'CircularGauss':
-            self.gk = get_sift_weighting_kernel(ksize=patch_size,
-                                                circ=True,
-                                                sigma_type=sigma_type)
-        elif self.mask_type == 'Gauss':
-            self.gk = get_sift_weighting_kernel(ksize=patch_size,
-                                                circ=False,
-                                                sigma_type=sigma_type)
-        elif self.mask_type == 'Uniform':
-            self.gk = torch.ones(patch_size,patch_size).float() / float(patch_size**2)
-        else:
-            raise ValueError(masktype, 'is unknown mask type. Try Gauss, CircularGauss or Uniform')
-            
-        self.bin_ksize, self.bin_stride, self.pad = get_sift_bin_ksize_stride_pad(patch_size, num_spatial_bins)
-        self.gx =  nn.Conv2d(1, 1, kernel_size=(1,3),  bias = False)
-        self.gx.weight.data = torch.tensor([[[[-1.0, 0.0, 1.0]]]]).float()
-        
-        self.gy = nn.Conv2d(1, 1, kernel_size=(3,1),  bias = False)
-        self.gy.weight.data = torch.tensor([[[[-1], [0], [1]]]]).float()
-        
-        nw = get_sift_pooling_kernel(ksize = self.bin_ksize).float()
-        
+
+        ks: int = self.patch_size
+        sigma: float = float(ks) / math.sqrt(2.0)
```

Comment:
maybe is a bit nitpick, but we can try to use pure torch for basic stuff ? just for convention with the library, no more arguments for it.